### PR TITLE
Change mod options and add example mod

### PR DESCRIPTION
**Pull request by AlexejheroYTB**

### CODE REVIEW

**Comment by PrimeSonic** [comment]
Looked good so far. I'll try this out when I get the chance.  
If you can write some sample test code for all these like we had before, it would be helpful, as well as serving as useful documentation.

File: SMLHelper/Options/ModOptions.cs [hunk 1]
```diff
@@ -312,4 +427,48 @@ internal ModChoiceOption(string id, string label, string[] options, int index) :
...
+        /// <param name="id">The internal ID of this option.</param>
+        /// <param name="label">The display text to show on the in-game menus.</param>
+        /// <param name="options">The collection of available values.</param>
+        /// <param name="index">The starting value.</param>
+        internal ModDropdownOption(string id, string label, string[] options, int index) : base(ModOptionType.Dropdown, label, id)
+        {
+            Options = options;
+            Index = index;
+        }
+    }
```
Comment (PrimeSonic) [inline SMLHelper/Options/ModOptions.cs]: This file is starting to get big, with too many barely related things contained.  
Let's split this up into distinct files for each class.

**Comment by ahk1221** [comment]
Looks good. As for what's going to happen with the choices/dropdown's options, I think maybe storing objects instead of strings, and then calling ToString on said object when building the options menu would be the best idea. Maybe could work with generics as well. And if you can get that custom wrapper to work, that would be awesome, so that we can get the actual object changed/selected instead of having to manually do that.

**Comment by AlexejheroYTB** [comment]
I am currently thinking about adding the following overloads to the choice and dropdown options:

    public static void AddChoiceOption(string id, string name, object[] options, int choice)
    public static void AddChoiceOption(string id, string name, object[] options, object value)
    public static void AddChoiceOption(string id, string name, string[] options, string value)
    public static void AddChoiceOption<T>(string id, string name, T value) where T : Enum
    public static void AddChoiceOption<T>(string id, string name, int choice) where T : Enum

File: SMLHelper/Options/ModOptions.cs (hunk 1 above)
Comment (AlexejheroYTB) [reply SMLHelper/Options/ModOptions.cs]: Addressed in commit 20c314b

**Comment by PrimeSonic** [comment]
If no one else does, I can add documentation on the remaining public classes and methods that don't have any.

**Comment by AlexejheroYTB** [comment]
> If no one else does, I can add documentation on the remaining public classes and methods that don't have any.

I added docs for all of the classes that I created. I didn't know there were others. I will try to find and document them myself

**Comment by AlexejheroYTB** [comment]
I think that documenting public members should be done in a different pr.

**Comment by AlexejheroYTB** [comment]
@ahk1221 This PR is ready to be merged IMO

**Pull request merged**